### ISSUE TITLE
Set up a Socket Server + Socket Client

### ISSUE THREAD

**Issue by MrPowerGamerBR** [PocketDreams/SequinLand#8]
This isn't related to SequinLand, but I'm creating a issue here anyway.

Due to the changes in 1.0.3.0, using proxy programs like Nemisys isn't needed anymore. (yay!)

However if you don't have a program like Nemisys, how would you communicate between Nukkit instances?

**TODO:** Create something like that, shouldn't be difficult.

**Comment by thebigsmileXD** [comment]
The client doesn't send that it was transfer, its simply a clean connect. Thought about it in our group yesterday, https://telegram.me/joinchat/AlErxD3AN_GOgeSAcS0twA, we got the idea to use a query (if you can create custom queries), or simply log into an DB that the player got transferred (would only work if both servers have access to the DB)

**Comment by MrPowerGamerBR** [comment]
@thebigsmileXD it would be better using sockets in my opinion, I already have something like that for my BungeeCord server (because I don't like using Bungee Plugin Messaging channel) and it works fine.

It would be like this:
```
        NoNameYet.getAllPlayers(60800, new NoNameYetCallback() {

            @Override
            public void receivedData(DataWrapper arg0) {
                // Data data data...
            }
            
        });
```

**Comment by MrPowerGamerBR** [comment]
The problem is... well, it would only work with Nukkit. (or you could implement it in a PHP plugin, AFAIK you can connect a PHP socket to a Java socket)

**Comment by thebigsmileXD** [comment]
Ah i see, you want to make it Java based only?The function seems cool already, but why would it only work with Nukkit?

**Comment by MrPowerGamerBR** [comment]
@thebigsmileXD because I'm too lazy to make a PocketMine-MP plugin 😉 

But in theory it can work with every server software (but the Socket Server will be in Java), as long as you implement a way to connect to the Socket Server in *insert language here*. (and as far as I know, you can connect a PHP socket client to a Java socket server, I already done that before)

**Comment by DaMatrix** [comment]
So would all clients be connected to each other or would there be one "master" server that manages chat while the other server connect to that one for chat stuff?

**Comment by MrPowerGamerBR** [comment]
@DaMatrix it would be one "master" server (which wouldn't be a MCPE server, it would be a socket server) that all the "slaves" (MCPE servers (Nukkit)) connects to receive info from other servers.

Not that complex and very easy to create.

**Comment by DaMatrix** [comment]
I'd be willing to do that, but would the master server be in the normal program (launch argument or something) or a separate branch / repo?

**Comment by MrPowerGamerBR** [comment]
@DaMatrix let me explain how it would be done:

You have some servers (in this case, Nukkit servers)
Lobby, Survival, Prison.

Then you have a "master" program, which would be the connection "hub" so those servers can get all players, transfer users from other servers to new servers, etc.

And you also have a "Connect" plugin, which you add to every Nukkit server to connect to the "master" program.

So, for example, you want to get all the players connected to the network from the "Survival" server, you use this:

```
        NoNameYet.getAllPlayers(60800, new NoNameYetCallback() {

            @Override
            public void receivedData(DataWrapper arg0) {
                // Data data data...
            }
            
        });
```

Which will actually do this:
The plugin sends a request to the "master" program
The "master" program requests the player count to all servers
All servers reply with the player count
The "master" program sends to the Survival server the player count.

Seems complicated but it is very easy to create.

Also, it won't be integrated in SequinLand, it will be a separate repo.

**Comment by MrPowerGamerBR** [comment]
https://github.com/PocketDreams/PocketHub

**Comment by MrPowerGamerBR** [comment]
Of course, the repo is empty because I didn't create the program (and the plugin) yet, and I will first test on my server before saying "hey, this is stable, go use it, nothing will explode!"

**Comment by thebigsmileXD** [comment]
Oh btw, i finally understand sockets and how to read them ;) i made my server check if the transferserver target server is online before attempting to connect

**Comment by MrPowerGamerBR** [comment]
@thebigsmileXD it was going to be easy to create a socket server like that until I remembered that I needed to create a JAR loader (for plugins)
